### PR TITLE
Feature/ARSN-73 Migrate source files to TypeScript

### PR DESCRIPTION
**Pull request by ghivert**
Acceptance criteria

* TS source files only
* Source maps generated
* JS application uploaded to artifacts

Compile/Upload CI step: https://github.com/scality/Arsenal/runs/5334705044?check_suite_focus=true
Artifacts: https://artifacts.scality.net/builds/github:scality:Arsenal:staging-5d5393bf4e.tests.644.1/

Vault CI result with Arsenal HEAD: https://eve.devsca.com/github/scality/vault/#/builders/2/builds/7377

Patchset required in Vault:

```diff
diff --git a/Dockerfile b/Dockerfile
index 180536f9..d26f5663 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -18,7 +18,7 @@ COPY ./yarn.lock /vault/
 WORKDIR /vault
 
 RUN apt-get update \
-    && apt-get install -y --no-install-recommends apt-utils python3  build-essential \
+    && apt-get install -y --no-install-recommends apt-utils git python3  build-essential \
     && yarn cache clean \
     && yarn install \
     && yarn cache clean \
diff --git a/package.json b/package.json
index ee37115a..40e2ce79 100644
--- a/package.json
+++ b/package.json
@@ -15,7 +15,7 @@
   "homepage": "https://github.com/scality/Vault#readme",
   "dependencies": {
     "@hapi/joi": "^17.1.0",
-    "arsenal": "scality/Arsenal#8.1.27",
+    "arsenal": "git://github.com/scality/Arsenal.git#df4b2e8eae5ead19948dd4326b9f72af60610216",
     "async": "1.5.2",
     "aws-sdk": "^2.754.0",
     "backo": "^1.1.0",

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,18 @@ In order to contribute, please follow the
...
 https://github.com/scality/Guidelines/blob/master/CONTRIBUTING.md).
 
+## Installation
+
+Because Arsenal is written in TypeScript, you need to pull the repository
+directly from GitHub with a git URL in the `package.json`. You need to setup
+Arsenal by adding this line to your `package.json`:
+
+```json
+"arsenal": "git://github.com/scality/Arsenal.git#[the-commit-hash-here]",
```
Comment (miniscruff) [inline README.md:33]: Don't we want to use tags now?

File: index.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
+export * as auth from './lib/auth/auth';
```
Comment (miniscruff) [inline index.ts:1]: Is this file right? It has a lot of require statements still.

**Review by miniscruff — COMMENTED** [review]
Just so many changes... about half way done with the files...

**Comment by ghivert** [comment]
Finally we ended up with a different strategy and we are maintaining 7.10 also. This PR should not exist anymore.

**Pull request closed without merging**